### DISCUSSION TITLE
Can't load new images

### DISCUSSION THREAD

**Discussion by agusadjadj** [mattieserver/netbox-topology-views#296] [Q&A]
Hi there!

I am trying to install new images to use in the plugin for Fortinet devices.

My NetBox is running in Docker, with its respective Dockerfile-Plugins and extra configurations working. The plugin it is working well.
I'm a bit confused where I should install these images, since by default they are in "'static_image_directory':'netbox_topology_views/img'"

I can't find that folder in the "nearby" of Docker, and I wouldn't know where to point that directory.

I was trying to do a "find" on my Linux, but the folders are in something that looks like a temp folder (var/lib/docker/overlay2/sea1riyr1zhevm10ocs0kbuit/diff/opt/netbox/venv/lib/python3.10/site-packages /netbox_topology_views/static/netbox_topology_views/img). This is a example, i have like 4-5 paths like that.

I would be grateful for how to make a modification to the static_image_directory within Docker.

Thank you very much in advance!

**Comment by dreng** [comment]
Sorry, can't help with Docker at all. Do you see any icons at all, maybe the old ones? You may try a "find" with one of these filenames then, for example "role-unknown.png". You may have a look here regarding the old filenames: https://github.com/mattieserver/netbox-topology-views/tree/v3.3.0/netbox_topology_views/static/netbox_topology_views/img

**Comment by mattieserver** [comment]
can you check `/opt/netbox/netbox/static/netbox_topology_views/img` ? this should be the folder where the images are stored

**Comment by agusadjadj** [comment]
The icons are being displayed correctly, without any problem at the moment. The strange thing is that the folder `/opt/netbox/netbox/static/netbox_topology_views/img` does not exist.
I'm deploying Netbox on a test server, just to solve these problems before going to production.
Before using Docker, I tried to install it manually, perhaps some strange dependency was left, or a configuration running that is not allowing me to move forward.
I'll try to install it from scratch again and see if I can fix it.

In response to the find, I can find any image, but the search returns the following:

`root@ubntsrv:~# find / -name role-unknown.png`
`/var/lib/docker/overlay2/l25t2wfervmp51jkt6m6uxdep/diff/opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/v5m12krygfjze3iwhwaefzoiq/diff/opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/wh5mufdfgrwz7k8fguuozvyaq/diff/opt/netbox/netbox/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/opnds7g5rqv4nnqt5qm5ww2kz/diff/opt/netbox/netbox/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/sea1riyr1zhevm10ocs0kbuit/diff/opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/jd53uowri4fecxp7ydhgxuihn/diff/opt/netbox/netbox/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/o3j8fulmp4z9dewkpalfpcjdi/diff/opt/netbox/netbox/static/netbox_topology_views/img/role-unknown.png`

`/var/lib/docker/overlay2/ox3111opcpkpgwgrsfvuw40q2/diff/opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views/static/netbox_topology_views/img/role-unknown.png`

Particularly it is my first deploy with Docker, so, probably, there is something that is escaping my knowledge and lack of experience. I will wait for comments! Thanks

**Comment by mattieserver** [comment]
Can you check if it works if you add the following to you dockerfile:
`
RUN mkdir -p /opt/netbox/netbox/static/netbox_topology_views/img
`
Should be before the `/opt/netbox/netbox/manage.py collectstatic --no-input` command (but after the install command)

**Comment by agusadjadj** [comment]
It doesn't seem like it's creating the folder, don't know why exactly.
I tried creating the folder myself, but it doesn't work as expected either.

However, I think I have an installation problem, since I did tests on this server and perhaps something was "misconfigured", which may be bothering collectstatic or something similar.

I will continue testing and try to reinstall, I will be updating as soon as I have any news.

**Comment by agusadjadj** [comment]
Fixed: I made the following changes to add external images. There is still room for improvement, but these were my steps.

1) I downloaded all the images from the repository `(netbox_topology_views/img)` to a folder where I have my Dockerfile.
2) In that folder, I add my .csv files (I haven't set max/min sizes yet, but I will soon).
3) Add to my docker-compose.yml , in the volumes part, the following: `- ./topology_views:/opt/netbox/netbox/static/netbox_topology_views/img:z,rw `, mapped the folder to the docker container.
4) I gave 777 permissions to the folder I created, I still have to test how it works without so many permissions.
5) Worked. Add .csv to that folder, add the images to the plugin settings!

**Comment by SweetSophia** [comment]
Since this happened to me too, I am going to revive this:
This is a collectstatic/static build issue, not a missing package issue. Try
/opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py collectstatic --no-input
or /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py collectstatic --no-input inside the container
files are located here: 
`/opt/netbox/venv/lib/python3.12/site-packages/netbox_topology_views/static/netbox_topology_views/
`
but should be here
`/opt/netbox/netbox/static/netbox_topology_views/
`
After that, check:
`ls -la /opt/netbox/netbox/static/netbox_topology_views`

`ls -la /opt/netbox/netbox/static/netbox_topology_views/img`
